hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -727,6 +727,21 @@ private class Emit(
       case _: ArrayMap | _: ArrayZip | _: ArrayFilter | _: ArrayFlatMap | _: ArrayScan | _: ArrayLeftJoinDistinct | _: RunAggScan | _: ReadPartition | _: MakeStream | _: StreamRange =>
         emitArrayIterator(ir).toEmitTriplet(mb, PArray(coerce[PStreamable](ir.pType).elementType))
 
+      case ArrayZeros(length) =>
+        val lengthTriplet = emit(length)
+        val outputPType = coerce[PArray](ir.pType)
+        val elementSize = outputPType.elementByteSize
+        val numElements = mb.newField[Int]
+        val arrayAddress = mb.newField[Long]
+        val result = Code(
+          numElements := lengthTriplet.value[Int],
```

Comment:
out of curiosity, do you need to check that length isn't missing here (.m)?

Reply:
Should be fine, since I return the `lengthTriplet.m` as my missingness value in the triplet. So no one should evaluate `result` if `lengthTriplet.m`is true.